# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/RangeUtils.h
# Commit: fb611166c7ab
# Full Hash: fb611166c7ab7d3c67f2361ec8c45e724d2e6dcc
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-05-09 04:15:00
# Regressor Bug: 1932150
# File Overlap Count: 2
# Description:
#   Bug 1932150 - Fix some inconsistency about indicating whether the selection is allowed to cross the shadow boundary r=jjaschke,dom-core
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D246949
# ==============================================================================

diff -r 53dc4e16ebe8 -r fb611166c7ab dom/base/RangeUtils.h
--- a/dom/base/RangeUtils.h	Thu May 08 12:56:29 2025 +0000
+++ b/dom/base/RangeUtils.h	Thu May 08 13:37:20 2025 +0000
@@ -25,23 +25,29 @@
 struct ShadowDOMSelectionHelpers {
   ShadowDOMSelectionHelpers() = delete;
 
-  static nsINode* GetStartContainer(const AbstractRange* aRange,
-                                    bool aAllowCrossShadowBoundary);
+  static nsINode* GetStartContainer(
+      const AbstractRange* aRange,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 
-  static uint32_t StartOffset(const AbstractRange* aRange,
-                              bool aAllowCrossShadowBoundary);
+  static uint32_t StartOffset(
+      const AbstractRange* aRange,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 
-  static nsINode* GetEndContainer(const AbstractRange* aRange,
-                                  bool aAllowCrossShadowBoundary);
+  static nsINode* GetEndContainer(
+      const AbstractRange* aRange,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 
-  static uint32_t EndOffset(const AbstractRange* aRange,
-                            bool aAllowCrossShadowBoundary);
+  static uint32_t EndOffset(
+      const AbstractRange* aRange,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 
-  static nsINode* GetParentNodeInSameSelection(nsINode& aNode,
-                                               bool aAllowCrossShadowBoundary);
+  static nsINode* GetParentNodeInSameSelection(
+      const nsINode& aNode,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 
-  static ShadowRoot* GetShadowRoot(const nsINode* aNode,
-                                   bool aAllowCrossShadowBoundary);
+  static ShadowRoot* GetShadowRoot(
+      const nsINode* aNode,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 };
 }  // namespace dom
 